### PR TITLE
Add support to  resolve CN in CEL expression

### PR DESCRIPTION
**Pull request by v-rosa**
The goal is to be able to create a matcher for client certificate DN using CEL expressions, for instance:

```
	@client_cert {
		expression {http.request.tls.client.subject} == 'CN=bla,OU=blabla,O=bla,L=bla,ST=bleh,C=buh'
	}
```

### CODE REVIEW

**Comment by francislavoie** [comment]
FYI, the Caddyfile offers a shortcut to that placeholder `{tls_client_subject}` https://caddyserver.com/docs/caddyfile/concepts#placeholders

**Comment by francislavoie** [comment]
I think this looks good with the `pkix.Name` wrapper! What do you think is remaining to do here?

File: modules/caddyhttp/celmatcher.go [hunk 1]
```diff
@@ -175,8 +176,19 @@ func (m MatchExpression) caddyPlaceholderFunc(lhs, rhs ref.Val) ref.Val {
...
 
+// MapStructToDefaultType Maps a struct to a CEL default type
+// func MapStructToDefaultType(value interface{}) interface{} {
+// 	switch value.(type) {
+// 	case pkix.Name:
+// 		return value.(pkix.Name).String()
+// 	default:
+// 		return value
+// 	}
+// }
```
Comment (mholt) [inline modules/caddyhttp/celmatcher.go:190]: Remove unused code

File: modules/caddyhttp/celmatcher.go [hunk 2]
```diff
@@ -175,8 +176,19 @@ func (m MatchExpression) caddyPlaceholderFunc(lhs, rhs ref.Val) ref.Val {
 	val, _ := repl.Get(string(phStr))
 
 	return m.ta.NativeToValue(val)
+	// return m.ta.NativeToValue(MapStructToDefaultType(val))
```
Comment (mholt) [inline modules/caddyhttp/celmatcher.go:179]: Remove unused code

File: modules/caddyhttp/celmatcher.go (hunk 1 above)
Comment (v-rosa) [reply modules/caddyhttp/celmatcher.go]: Done.

File: modules/caddyhttp/celmatcher.go (hunk 2 above)
Comment (v-rosa) [reply modules/caddyhttp/celmatcher.go]: Done

**Comment by mholt** [comment]
Great, thank you for the contribution!

**Pull request merged**